### PR TITLE
rgw/multisite: Don't allow certain radosgw-admin commands to run on non-master zone

### PR DESCRIPTION
**Pull request by smanjara**
These are commands that modify metadata. Can override with 'yes-i-really-mean-it' option to allow the changes only on the local zone. 

Fixes: http://tracker.ceph.com/issues/39548

Signed-off-by: Shilpa Jagannath <smanjara@redhat.com>

- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

File: src/rgw/rgw_admin.cc [hunk 1]
```diff
@@ -6406,6 +6437,11 @@ int main(int argc, const char **argv)
   }
 
   if (opt_cmd == OPT_OBJECT_UNLINK) {
+    if (!store->svc.zone->is_meta_master() && !yes_i_really_mean_it) {
```
Comment (cbodley) [inline src/rgw/rgw_admin.cc]: it doesn't look like OPT_OBJECT_UNLINK touches any metadata

File: src/rgw/rgw_admin.cc [hunk 2]
```diff
@@ -4996,6 +4996,12 @@ int main(int argc, const char **argv)
     }
     break;
   case OPT_USER_CREATE:
+    if (!store->svc.zone->is_meta_master() && !yes_i_really_mean_it) {
+      cerr << "user created here will not be synced to master zone" << std::endl;
+      cerr << "do you really mean it? (requires --yes-i-really-mean-it)" << std::endl;
```
Comment (cbodley) [inline src/rgw/rgw_admin.cc]: we should also strongly recommend running the command on the master zone instead

**Comment by smanjara** [comment]
_> this also needs to cover the write commands for METADATA, SUBUSER, OTP and RESHARD._ 

Yes, I'm yet to add those.

_>i think it's probably better to do this check in one place at the top, instead of duplicating the check for each command. can you add something similar to `raw_storage_ops_list` and `readonly_ops_list` for this?_

Thanks for the pointer. I was not sure how to minimize duplication. This helps.

**Comment by smanjara** [comment]
@cbodley, can you please review again? Since we don't support dynamic resharding with multisite, I don't think we need to do the check there.

**Comment by smanjara** [comment]
> it helps, but i still worry that we'll forget to update this when adding new commands. can you make sure that bucket chown and user rename end up here?

Sure, will do.

**Comment by alimaredia** [comment]
PR is passing on wip-ali-testing run here: http://pulpito.ceph.com/amaredia-2019-07-16_16:09:11-rgw-wip-ali-testing-distro-basic-smithi/

**Pull request merged**